code/modules/atmospherics/machinery/pipes/smart.dm

```diff
@@ -56,6 +61,17 @@ GLOBAL_LIST_INIT(atmos_components, typecacheof(list(/obj/machinery/atmospherics)
 					bitfield |= WEST_SHORTPIPE
 
 	icon_state = "[bitfield]_[piping_layer]"
+	switch(connection_num)
+		if(0)
+			dir = 2
+		if(1)
+			dir = connections
+		if(2)
+			dir = check_binary_direction(connections)
+		if(3)
+			dir = connections
+		if(4)
+			dir = 15
```

Comment:
What's the point of this?

Reply:
Bridge pipes apparently could only be placed on pipes facing 1 direction, even if the 2 are on opposite sides. Not sure if it was intentional or not, so I added this check.

Reply:
Magic numbers are bad, use defines

Reply:
Maybe I could make bridge (and similar) pipes check for directions facing both ways if you dislike the switch statement. Not going to change anything right now.

Reply:
when you say facing one direction you mean like, just going north right?

Reply:
Yeah. From earlier testing, a horizontal bridge pipe could only be placed on a pipe facing north or south, but not both.